SearchImplem/GoogleAutocomplete.swift

```diff
@@ -73,14 +73,75 @@ class GoogleAutocomplete: UIViewController, UITableViewDataSource, UITableViewDe
                     }
                     for item in placePredictions {
                         let placeDescription = item["description"] as! String
-                        self.predictions.append(placeDescription)
+                        let googleID = item["place_id"] as! String
+                        let separatedStrings = placeDescription.components(separatedBy: ", ")
+                        let city = City()
+                        city.googleLocationID = googleID
+                        self.getPlaceDetails(id: city.googleLocationID!)
+                        switch separatedStrings.count {
+                        case 1:
+                            city.name = separatedStrings[0]
+                        case 2:
+                            city.name = separatedStrings[0]
+                            city.country = separatedStrings[1]
+                        case 3:
+                            city.name = separatedStrings[0]
+                            city.adminArea = separatedStrings[1]
+                            city.country = separatedStrings[2]
+                        default:
+                            break
+                        }
+                        self.predictions.append(city)
                     }
                     DispatchQueue.main.async {
                         self.tableView.reloadData()
                     }
                 }
                 catch let error {
-                    // handle error
+                    print(error)  // handle error
+                }
+            }
+        }
+        task.resume()
+    }
+    func getPlaceDetails(id: String) {
+        let apiKey:String = "AIzaSyAj4WP8kwzjstnDvWqGMNehJ9YNSxIh_Fs"
```

Comment:
remove your API Key from code. maybe find a way to automate the process if you use the app and let others manually add their API Key.